web/src/main/java/ua/com/vertex/logic/UserLogicImpl.java

```diff
@@ -66,12 +69,34 @@ public void saveImage(String email, MultipartFile file, String imageType) {
     }
 
     private void validateMultipartFile(MultipartFile file) {
-        if (file.getSize() > fileSizeInBytes) {
-            throw new MultipartValidationException("File size exceeded, max allowed is " +
-                    UtilFunctions.humanReadableByteCount(fileSizeInBytes));
+        if (!file.isEmpty()) {
+            if (file.getSize() > fileSizeInBytes) {
+                throw new MultipartValidationException("File size exceeded, max allowed is " +
+                        UtilFunctions.humanReadableByteCount(fileSizeInBytes));
+            } else if (!file.getContentType().split("/")[0].equals("image")) {
+                throw new MultipartValidationException("You have chosen a file of invalid type");
+            }
+        }
+    }
+
+    @Override
+    public boolean validateMultipartFileWithBindingResult(MultipartFile file, BindingResult result, String image) {
+        boolean validation = true;
+        if (file.isEmpty()) {
+            validation = false;
+
+        } else if (file.getSize() > fileSizeInBytes) {
+            result.rejectValue(image, "error." + image,
+                    FILE_SIZE + UtilFunctions.humanReadableByteCount(fileSizeInBytes));
+            LOGGER.debug("Image size invalid: too big");
+            validation = false;
+
         } else if (!file.getContentType().split("/")[0].equals("image")) {
-            throw new MultipartValidationException("You have chosen a file of invalid type");
+            result.rejectValue(image, "error." + image, FILE_TYPE);
+            LOGGER.debug("Image type invalid: not image");
```

Comment:
please add actual result

Reply:
'not image', but what? you are expecting file.getContentType().split("/")[0].equals("image") so can you please what was actual result in case it is not image?